test_plans/atomic_ref.asciidoc

```diff
@@ -505,4 +507,137 @@ Check if it returns the original value of the referenced object. Check returned
 Check if it computes the maximum of `operand` and the value of the object referenced by this `atomic_ref`,
 and assigns the result to the value of the referenced object.
 If `T` is `Floating` check that new value of the referenced object is equal to `(maximum value + operand) +- epsilon`.
-Check if it returns the original value of the referenced object. Check returned value type.
\ No newline at end of file
+Check if it returns the original value of the referenced object. Check returned value type.
+
+=== Stress tests
+
+==== Atomicity for device scope
+
+For the following values as `order` parameter:
+
+* `memory_order::relaxed`
+* `memory_order::acq_rel`
+* `memory_order::seq_cst`
+
+For the following values as `scope` parameter:
+
+* `memory_scope::device`
+* `memory_scope::system`
+
+And the following values as `space` template parameter:
+
+* `access::address_space::global_space`
+* `access::address_space::generic_space`
+
+Create `int` variable `val` and initialize it with `0`.
+Create `sycl::accessor acc` for `val` with `target` = `device`, `access_mode` = `read_write`.
+Call `parallel_for` with range<1> parameter
+Create `sycl::atomic_ref<int, order, scope, space> a_r` for `acc[0]`.
+Call `a_r.fetch_add(2)`.
+After kernel is executed check that `val` = `range.size() * 2`.
```

Comment:
Can you rewrite these descriptions to show a small code sample that illustrates the core part of the test?  For example, this first one might be something like:

> This test uses a simple-form `parallel_for` to atomically update a single scalar value from all work-items:
>
> ```
> TYPE val{};
> sycl::buffer buf{&val, {1}};
> q.submit([&](sycl::handler &cgh) {
>   sycl::accessor acc{buf};
>   cgh.parallel_for({SIZE}, [=](int i) {
>      sycl::atomic_ref<TYPE, ORDER, SCOPE, SPACE> a_r{acc[0]};
>      a_r.fetch_add(2);
>   });
> });
> ```
>
> Verify that `val == SIZE*2` after the loop completes.  (The result must be accurate within some small error range when `TYPE` is a floating point type.)
>
> This test is repeated for all combinations of supported values listed below:
>
> These `TYPE` types:
>
> * `int`
> * `long long` (if `sizeof(long long) == 8` then only if the device has `aspect::atomic64`)
> * `double` (only if the device has `aspect::fp64` and if `sizeof(double) == 8` only if the device has `aspect::atomic64`)
>
> These `SCOPE` values which are supported by the device (according to `info::device::atomic_memory_scope_capabilities`):
>
> * `memory_scope::device`
> * `memory_scope::system`
>
> These `ORDER` values which are supported by the device (according to `info::device::atomic_memory_order_capabilities`):
>
> * `memory_order::relaxed`
> * `memory_order::acq_rel`
> * `memory_order::seq_cst`
>
> These `SPACE` values:
>
> * `access::address_space::global_space`
> * `access::address_space::generic_space`

We used this type of test description in the [atomic_fence](https://github.com/KhronosGroup/SYCL-CTS/blob/SYCL-2020/test_plans/atomic_fence.asciidoc) test plan, and I think it makes the plan much easier to understand.

Note that I also suggest some additional types for this test.  I think it's important to test a 64-bit type and a floating point type also.

Reply:
Rewrote test cases description to this format, added also type `float` to test floating point type even if fp64 is not supported.
Also changed how atomicity with respect to atomic operations in host code should be tested to correctly deal with pointers.